terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/scale-down.ts

```diff
@@ -225,6 +239,46 @@ export async function scaleDown(): Promise<void> {
   }
 }
 
+export async function cleanupOldSSMParameters(runnersRgions: Set<string>, metrics: ScaleDownMetrics): Promise<void> {
+  try {
+    for (const awsRegion of runnersRgions) {
+      const ssmParams = sortSSMParaametersByUpdateTime(
+        Array.from((await listSSMParameters(metrics, awsRegion)).values()),
+      );
+
+      let deleted = 0;
+      for (const ssmParam of ssmParams) {
+        /* istanbul ignore next */
+        if (ssmParam.Name === undefined) {
+          continue;
+        }
+        if (ssmParam.LastModifiedDate === undefined) {
+          break;
+        }
+        if (
+          ssmParam.LastModifiedDate.getTime() >
+          moment().subtract(Config.Instance.sSMParamCleanupAgeDays, 'days').toDate().getTime()
+        ) {
+          break;
+        }
+        if (await doDeleteSSMParameter(ssmParam.Name, metrics, awsRegion)) {
+          deleted += 1;
+        }
+        if (deleted >= Config.Instance.sSMParamMaxCleanupCount) {
```

Comment:
Looks like `sSMParamMaxCleanupCount` is the max number of params that should get cleaned up on a single scale down request.  However, when ten instances are being scaled down one after another, you could end up deleting 10 * `sSMParamMaxCleanupCount` instances by the end.

Is that right?

If so, I'm wondering if we can change this name to reflect that expectation.  For context, my first assumption was that this var represented the threshold where we would only cleanup SSM params if we had more than `sSMParamMaxCleanupCount` entries

Potential new name: sSMParamCleanupAllowancePerScaledown

Reply:
We could, but I really dislike those variable names that are a phrase and very long.
Can we come up with a name that is shorter? `sSMParamMaxCleanupAllowance`? 
It is a bit too high for my taste, but I am kind of OK with that.

Having said that, can you explain what you mean by:
```
Looks like sSMParamMaxCleanupCount is the max number of params that should get cleaned up on a single scale down request. However, when ten instances are being scaled down one after another, you could end up deleting 10 * sSMParamMaxCleanupCount instances by the end.
```

Reply:
Sure, `sSMParamMaxCleanupAllowance` works

> Having said that, can you explain what you mean by:
The parameter's functionality just surprised me.  It seems like the intention was to limit how much damage a single scale down request can do, but when there are multiple scale down requests rouge code can still do lots of damage.

Or perhaps the intention here was to limit the max duration of a scale down request? That would make more sense.

Reply:
That is the intention, I did review the code and it seems it is doing as planned.

It limits the impact for `sSMParamMaxCleanupCount` by run by AWS region (we only have 1 region atm, as multi-region is a WIP)